api/api/views/media_views.py

```diff
@@ -17,6 +21,62 @@
 logger = logging.getLogger(__name__)
 
 
+class AsyncMediaView(ViewSet, GenericViewSet):
+    """
+    Encapsulate asynchronous media requests.
+
+    Isolates asynchronous API routes from yet-to-be-converted
+    synchronous routes.
+    """
+
+    view_is_async = True
+
+    lookup_field = "identifier"
+    lookup_value_regex = (
+        r"[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}"
+    )
+    model_class: type[AbstractMedia] = None
+
+    def get_queryset(self):
+        # The alternative to a sub-query would be using `extra` to do a join
+        # to the content provider table and filtering `filter_content`. However,
+        # that assumes that a content provider entry exists, which is not necessarily
+        # the case. We often don't add a content provider until after works from
+        # new providers are available in the API, and sometimes not even then.
+        # Search returns results with providers that do not have a ContentProvider
+        # table entry. Therefore, to maintain that assumption, a subquery is the only
+        # workable approach, as Django's `extra` does not provide any facility for
+        # handling null relations on the join.
+        return self.model_class.objects.exclude(
+            provider__in=ContentProvider.objects.filter(
+                filter_content=True
+            ).values_list("provider_identifier")
+        )
```

Comment:
This is duplicated from the existing media view. We can abstract it into a mixin if we want to avoid the duplication. I'd prefer to keep them separate in case we need to modify this for async context for some reason.